### PR TITLE
Add stackdriver tests

### PR DESCRIPTION
**Pull request by jjzeng-seattle**

### CODE REVIEW

File: metrics/resource_view_test.go [hunk 1]
```diff
@@ -147,6 +152,50 @@ func (m metricExtract) String() string {
...
+	}
+	credentialsContent := []byte(`{"type": "service_account"}`)
+	if _, err := tmp.Write(credentialsContent); err != nil {
+		return err
+	}
+	os.Setenv("GOOGLE_APPLICATION_CREDENTIALS", tmp.Name())
+	return nil
+}
+
+func sortMetrics() cmp.Option {
```
Comment (evankanderson) [inline metrics/resource_view_test.go:180]: You can make this a `var` so you don't need to call the function, but I'm fine either way.

File: metrics/resource_view_test.go [hunk 2]
```diff
@@ -296,34 +330,18 @@ testComponent_testing_value{project="p1",revision="r2"} 1
...
 		},
 	}, {
 		name: "Stackdriver",
 		init: func() error {
-			if err := sdFake.start(); err != nil {
-				return err
-			}
-			conn, err := grpc.Dial(sdFake.address, grpc.WithInsecure())
+			var err error
+			err = initSdFake(&sdFake)
```
Comment (evankanderson) [inline metrics/resource_view_test.go]: Combine these three lines, like so:

```go
if err := initSdFake(&sdFake); err != nil {
  return err
}
```

File: metrics/resource_view_test.go [hunk 3]
```diff
@@ -378,11 +396,123 @@ testComponent_testing_value{project="p1",revision="r2"} 1
...
 		})
 	}
 }
 
+func TestStackDriverExports(t *testing.T) {
+	sdFake := stackDriverFake{address: "localhost:12346"}
+	eo := ExporterOptions{
+		Domain:         servingDomain,
+		Component:      "autoscaler",
+		PrometheusPort: 9090,
```
Comment (evankanderson) [inline metrics/resource_view_test.go]: Do you need to set `PrometheusPort`?

File: metrics/resource_view_test.go [hunk 4]
```diff
@@ -378,11 +396,123 @@ testComponent_testing_value{project="p1",revision="r2"} 1
...
+		"desired_pods",
+		"Number of pods that are desired",
+		stats.UnitDimensionless)
+	desiredPodsCountView := &view.View{
+		Description: "Number of pods that are desired",
+		Measure:     desiredPodCountM,
+		Aggregation: view.LastValue(),
+	}
+	err := initSdFake(&sdFake)
+	if err != nil {
```
Comment (evankanderson) [inline metrics/resource_view_test.go]: Combine again:
```suggestion
	if err := initSdFake(&sdFake); err != nil {
```

File: metrics/resource_view_test.go [hunk 5]
```diff
@@ -378,11 +396,123 @@ testComponent_testing_value{project="p1",revision="r2"} 1
...
+		t.Errorf("Init stackdriver failed %s", err)
+	}
+	err = UpdateExporter(eo, logtesting.TestLogger(t))
+	if err != nil {
+		t.Errorf("UpdateExporter failed %s", err)
+	}
+	sdFake.t = t
```
Comment (evankanderson) [inline metrics/resource_view_test.go]: Why not set this in the constructor?

File: metrics/resource_view_test.go [hunk 6]
```diff
@@ -378,11 +396,123 @@ testComponent_testing_value{project="p1",revision="r2"} 1
...
+
+	err = RegisterResourceView(desiredPodsCountView, actualPodsCountView)
```
Comment (evankanderson) [inline metrics/resource_view_test.go]: You don't do anything with this `err`

File: metrics/resource_view_test.go (hunk 2 above)
Comment (jjzeng-seattle) [reply metrics/resource_view_test.go]: Done.

File: metrics/resource_view_test.go (hunk 3 above)
Comment (jjzeng-seattle) [reply metrics/resource_view_test.go]: Done.

File: metrics/resource_view_test.go (hunk 4 above)
Comment (jjzeng-seattle) [reply metrics/resource_view_test.go]: Done.

File: metrics/resource_view_test.go (hunk 5 above)
Comment (jjzeng-seattle) [reply metrics/resource_view_test.go]: Done.

File: metrics/resource_view_test.go (hunk 6 above)
Comment (jjzeng-seattle) [reply metrics/resource_view_test.go]: Done.

**Review by evankanderson — APPROVED** [review]

**Pull request merged**